activerecord/test/cases/errors_test.rb

```diff
@@ -13,4 +15,19 @@ def test_can_be_instantiated_with_no_args
       end
     end
   end
+
+  def test_association_errors_copied_to_foreign_key
```

Comment:
Personally I don't think validations_test is the right place for it, as the logic relates more to errors itself than validations, but it's your call so I will modify it to reflect your requested changes.